StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+PaymentMethodAvailability.swift

```diff
@@ -132,5 +132,63 @@ extension PaymentSheet {
+        var debugDescription: String {
+            switch self {
+            case .unavailable:
+                return "unavailable: This payment method is not available."
+            case .returnURL:
+                return "returnURL: A return URL must be set, see https://stripe.com/docs/payments/accept-a-payment?platform=ios&ui=payment-sheet#ios-set-up-return-url"
+            case .shippingAddress:
+                return "shippingAddress: A shipping address must be present on the Intent or collected through the Address Element and populated on PaymentSheet.Configuration.shippingDetails. See https://stripe.com/docs/api/payment_intents/object#payment_intent_object-shipping and https://stripe.com/docs/elements/address-element/collect-addresses?platform=ios#ios-pre-fill-billing"
+            case .userSupportsDelayedPaymentMethods:
+                return "userSupportsDelayedPaymentMethods: PaymentSheet.Configuration.allowsDelayedPaymentMethods must be set to true."
+            case .financialConnectionsSDK:
+                return "financialConnectionsSDK: The FinancialConnections SDK must be linked."
+            case .validUSBankVerificationMethod:
+                return "validUSBankVerificationMethod: Requires a valid US bank verification method."
+            }
+        }
+
+    }
+
+    enum PaymentMethodAvailabilityStatus: Equatable {
+        /// This payment method is supported by PaymentSheet and the current configuration and intent
+        case supported
+        /// This payment method is not supported by PaymentSheet and/or the current configuration or intent
+        case notSupported
+        /// This payment method is not activated in live mode in the Stripe Dashboard
+        case unactivated
+        /// This payment method has requirements not met by the configuration or intent
+        case missingRequirements([PaymentMethodTypeRequirement])
+
+        var description: String {
+            switch self {
+            case .supported:
+                return "Supported by PaymentSheet."
+            case .notSupported:
+                return "Not currently supported by PaymentSheet."
+            case .unactivated:
+                return "Activated for test mode but not activated for live mode:. Visit the Stripe Dashboard to activate the payment method. https://support.stripe.com/questions/activate-a-new-payment-method"
+            case .missingRequirements(let missingRequirements):
+                return "\(missingRequirements.reduce("") { $0 + $1.debugDescription }) "
+            }
+        }
+
+        static func ==(lhs: PaymentMethodAvailabilityStatus, rhs: PaymentMethodAvailabilityStatus) -> Bool {
+            switch (lhs, rhs) {
+            case (.notSupported, .notSupported):
+                return true
+            case (.supported, .supported):
+                return true
+            case (.unactivated, .unactivated):
+                return true
```

Comment:
Nit: I don't have Xcode open, but if memory serves we can condense this to:

```suggestion
            case (.notSupported, .notSupported),
                 (.supported, .supported),
                 (.unactivated, .unactivated):
                return true
```